sync-core/src/main/java/com/cloudant/sync/query/Index.java

```diff
@@ -0,0 +1,165 @@
+
+    protected final Map<String, String> indexSettings;
+
+    private Index(List<Object> fieldNames,
+                  String indexName,
+                  String indexType,
+                  Map<String, String> indexSettings) {
+        this.fieldNames = fieldNames;
+        this.indexName = indexName;
+        this.indexType = indexType;
+        this.indexSettings = indexSettings;
+    }
+
+    public static Index setUp(List<Object> fieldNames, String indexName) {
+        // JSON index type is the default
+        return setUp(fieldNames, indexName, JSON_TYPE);
+    }
+
+    public static Index setUp(List<Object> fieldNames, String indexName, String indexType) {
+        return setUp(fieldNames, indexName, indexType, null);
+    }
+
+    /**
+     * The setUp method handles index specific validation and ensures that the constructed
+     * Index object is valid.
+     *
+     * @param fieldNames the field names in the index
+     * @param indexName the index name
+     * @param indexType the index type (json or text)
+     * @param indexSettings the optional settings used to configure the index.
+     *                      Only supported parameter is 'tokenize' for text indexes only.
+     * @return the Index object or null if arguments passed in were invalid.
+     */
+    public static Index setUp(List<Object> fieldNames,
+                              String indexName,
+                              String indexType,
+                              Map<String, String> indexSettings) {
+        if (fieldNames == null || fieldNames.isEmpty()) {
+            logger.log(Level.SEVERE, "No field names were provided.");
+            return null;
+        }
+
+        if (indexName == null || indexName.isEmpty()) {
+            logger.log(Level.SEVERE, "No index name was provided.");
+            return null;
+        }
+
+        if (indexType == null || indexType.isEmpty()) {
+            logger.log(Level.SEVERE, "No index type was provided.");
+            return null;
+        }
+
+        if (!validTypes.contains(indexType.toLowerCase())) {
+            logger.log(Level.SEVERE, String.format("Invalid index type %s.", indexType));
+            return null;
+        }
+
+        if (indexType.equalsIgnoreCase(JSON_TYPE) && indexSettings != null) {
+            logger.log(Level.WARNING, String.format("Index type is %s, index settings %s ignored.",
+                                                    indexType,
+                                                    indexSettings.toString()));
+            indexSettings = null;
+        } else if (indexType.equalsIgnoreCase(TEXT_TYPE)) {
+            if (indexSettings == null) {
+                indexSettings = new HashMap<String, String>();
+                indexSettings.put(TOKENIZE, DEFAULT_TOKENIZER);
+                logger.log(Level.INFO, String.format("Index type is %s, defaulting settings to %s.",
+                        indexType,
+                        indexSettings.toString()));
```

Comment:
There isn't, but when we converted to java.util.logging, I adopted using `Level.FINE` as the debug level